File: problem-solving/algorithms/java/HackerrankPsAlgorithms0002.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class HackerrankPsAlgorithms0002 {
	// Simple Array Sum
	// https://www.hackerrank.com/challenges/simple-array-sum/problem?isFullScreen=true

	// --- You need to fill this method ---
	public static int simpleArraySum(List<Integer> ar) {
		return ar.stream().mapToInt(Integer::intValue).sum();
	}

	public static void main(String[] args) throws IOException {
		BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

		int arCount = Integer.parseInt(bufferedReader.readLine().trim());

		List<Integer> ar = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
				.map(Integer::parseInt)
				.collect(Collectors.toList());

		int result = simpleArraySum(ar);

		bufferedWriter.write(String.valueOf(result));
		bufferedWriter.newLine();

		bufferedReader.close();
		bufferedWriter.close();
	}

}

File: problem-solving/algorithms/dotnet/HackerrankPsAlgorithms0002.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Result
{
	// Simple Array Sum
	// https://www.hackerrank.com/challenges/simple-array-sum/problem?isFullScreen=true

	public static int simpleArraySum(List<int> ar)
	{
		// --- You need to fill this method ---
		return ar.Sum();
	}

}

class Solution
{
	public static void Main(string[] args)
	{
		TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

		int arCount = Convert.ToInt32(Console.ReadLine().Trim());

		List<int> ar = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arTemp => Convert.ToInt32(arTemp)).ToList();

		int result = Result.simpleArraySum(ar);

		textWriter.WriteLine(result);

		textWriter.Flush();
		textWriter.Close();
	}
}
